src/pages/[...slug].astro

```diff
@@ -46,13 +125,24 @@ const extractUsername = (string:string) => {
         `}
     >
          <h1 class="m-0 mb-0">{entry.data.title}</h1> 
-        {entry.data.author !== "Anonymous" && (
+        {entry.data.author_data  && (
           <div class="flex items-center">
             <p class="mr-4">by:</p> 
-            <img src={`${entry.data.author}.png`} alt={"author"} width={25} height={25} class="mr-2" />
-            <a href={entry.data.author} target="_blank" rel="noopener noreferrer">
-              {extractUsername(entry.data.author)}
-            </a>
+
+            
+            <img src={ (entry.data.author_data as {pfp:string}).pfp } class="w-8 h-8 rounded-full mr-2" />
+
+            { (entry.data.author_data as {url:string}).url &&(
```

Comment:
The initial opening parentheses `(` was never closed, causing a syntax error. Please remove.